.github/workflows/v2.yml

```diff
@@ -93,10 +96,26 @@ jobs:
       - name: Check that fakes are updated
         run: git diff --no-ext-diff --exit-code
 
-
   full-ci:
     name: V2 CI Check (16.13.2, 1.17.5)
     runs-on: ubuntu-latest
     needs: [ci-go, ci-static, ci-js]
+    permissions:
+      contents: 'read'
+      id-token: 'write'
     steps:
+      - uses: actions/checkout@v3
+
+      - name: Authenticate to Google Cloud
+        id: gcloud-auth
+        uses: google-github-actions/auth@v0
+        with:
+          workload_identity_provider: ${{ secrets.workload_identity_provider }}
+          service_account: ${{ secrets.service_account }}
+      - uses: google-github-actions/setup-gcloud@v0
+      - run: gcloud --quiet auth configure-docker ${{ env.CI_CONTAINER_REGISTRY }}
+      - run: make dependencies
+      - name: Build & push docker iamges to Google
+        run: DOCKER_REGISTRY=${{ env.CI_CONTAINER_REPOSITORY }} make docker-image-push
```

Comment:
`make docker-image-push` actually re-tags the image with the git commit. I put this in the co-ordination job so that images are only pushed if tests & linting passes (both of which happen in the repo itself rather than in docker).

I think it makes sense to fix this but I also think that work is out of scope of this PR. A proposed fix would be replace the existing `ci-js` and `ci-go` jobs with `ci-cli` and `ci-server` (or whatever) jobs that:
* use docker build's `--target` flag to run the respective install depenencies steps
* then run tests inside the docker container
* complete building the various docker images
* use the [upload](https://github.com/marketplace/actions/upload-a-build-artifact) and [download](https://github.com/marketplace/actions/download-a-build-artifact) actions to pass these completed images to a final step
* push all the completed docker images to google only if everything finished (including the static check which I believe can remain unchanged).

this would, in theory, probably benefit from creating a 'base' docker image which has all the dependencies which is also used to run the unit tests against to avoid duplication of effort whilst sharing the work of downloading things. e.g.

```mermaid
graph TD;
    base-go[base-go: make dep, make unit-tests etc.]-->gitops[gitops: make gitops];
    base-go-->gitops-server-go;
    gitops-server-go[gitops-server-go: make -o index.html gitops-server]-->gitops-server[gitops-server: copy gitops-server ui]
    npm[npm: make ui]-->gitops-server;
    
```

Reply:
> make docker-image-push actually re-tags the image with the git commit

Gotcha. My mistake.

As for the "CI now takes 20 minutes" part, does it matter if tests pass? Nothing'll hit main without all the tests passing, and "can we build docker images" is a worthwile test chain on its own. When we want to fix it so that only successful PR pushes uploads an image, we can just use the upload/download actions between building the image and uploading it.

Reply:
I'm not sure? I think it makes sense to not push images that have failed their tests but can definitely gate that in the final stage.

Given our chat on slack I'm not against splitting this to build gitops, gitops-server and the images all in parallel with the unit tests and use up/download to gate the final push?